semantic/types.go

```diff
@@ -39,9 +43,44 @@ var natureNames = []string{
 	Function: "function",
 }
 
+var namesToNatures = map[string]Nature{
+	natureNames[Invalid]:  Invalid,
+	natureNames[String]:   String,
+	natureNames[Bytes]:    Bytes,
+	natureNames[Int]:      Int,
+	natureNames[UInt]:     UInt,
+	natureNames[Float]:    Float,
+	natureNames[Bool]:     Bool,
+	natureNames[Time]:     Time,
+	natureNames[Duration]: Duration,
+	natureNames[Regexp]:   Regexp,
+	natureNames[Array]:    Array,
+	natureNames[Object]:   Object,
+	natureNames[Function]: Function,
+}
+
 func (n Nature) String() string {
 	if int(n) < len(natureNames) {
 		return natureNames[n]
 	}
 	return "nature" + strconv.Itoa(int(n))
 }
+
+func (n Nature) MarshalJSON() ([]byte, error) {
+	s := n.String()
+	return json.Marshal(s)
+}
```

Comment:
These changes are no longer necessary, correct?

Reply:
Yes that's correct.